cmd/keymaster/main.go

```diff
@@ -140,16 +141,66 @@ func loadConfigFile(client *http.Client, logger log.Logger) (
 	return
 }
 
+func preConnectToHost(baseUrl string, client *http.Client, logger log.DebugLogger) error {
+	response, err := client.Get(baseUrl)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+	//we have to consume the contents of the body in order to keep the connection open
+	_, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		logger.Debugf(1, "Error reading http responseBody err: %s\n", err)
+		return err
+	}
+	if response.StatusCode >= 300 {
+		logger.Debugf(1, "bad response code on pre-connect status=%d", response.StatusCode)
+		return err
```

Comment:
These log entries should probably have higher level than debug?

Reply:
No, since we can have multiple servers we want to report errors only if none succeeds. The errors are actually collected by backgroundConnectToAnyKeymasterServer and printed if all of them fail.

Reply:
no, becasue we dont want to pollute the users console on non-fatal errors. Only when none of the keymasters can be contacted is when the error gets triggered and those are collected by the caller.